src/modal/test/modal.spec.js

```diff
@@ -833,6 +833,34 @@ describe('$uibModal', function () {
         expect($document.find('.modal-backdrop')).not.toHaveClass('fade');
       });
     });
+    
+    describe('container', function() {
+      it('should be added to body by default', function() {
+        var modal = open({template: '<div>Content</div>'});
+
+        expect($document).toHaveModalsOpen(1);
+        expect($document).toHaveModalOpenWithContent('Content', 'div');
+      });
+
+      it('should not be added to body if container is passed', function() {
+        var subElement = '<div clas="details-panel"></div>';
+        var element = angular.element(subElement);
+        angular.element(document.body).append(element);
+
+        var modal = open({template: '<div>Content</div>', container: '.details-panel'});
+
+        expect($document).not.toHaveModalOpenWithContent('Content', 'div');
```

Comment:
it is not sufficient to test that the body doesn't have a modal.  you need to check that the modal is on the component you've specified.